src/NUnitFramework/framework/Constraints/DictionaryContainsKeyConstraint.cs

```diff
@@ -231,7 +231,7 @@ private static MethodInfo FindContainsKeyMethod(Type type)
             var methods = type.GetMethods(BindingFlags.Instance | BindingFlags.Public);
             var method = methods.FirstOrDefault(m =>
                 m.ReturnType == typeof(bool)
-                && m.Name == "ContainsKey"
+                && (m.Name == "ContainsKey" || (m.Name == "Contains" && m.DeclaringType == typeof(IDictionary)))
```

Comment:
Since ` == typeof(IDictionary)` shows that you have this method in mind in particular:

```suggestion
                && (m.Name == "ContainsKey" || (m.Name == nameof(IDictionary.Contains) && m.DeclaringType == typeof(IDictionary)))
```